### PR TITLE
Google Ads Connector: Release community-contributed 

### LINKED ISSUE #3940: Release community-contributed Google Ads Connector
## Tell us about the problem you're trying to solve
#3842  introduced the Google Ads connector via contribution from the awesome @harshithmullapudi . There's a few loose ends we should tie up once that PR is merged: 

- Add docs to SUMMARY.md, public docs, build status pages. The instructions for getting the account credentials are the same as they are for google adwords. 
- update the contributor in #3869 since he is blocked on that PR
- Update the google adwords connector name in sourcedefinition YAML to add the `(Deprecated)` postfix. We should **not** remove the Google Adwords connector just yet. We still need to implement quite a number of reports and endpoints on Google Ads before we can move forward. 
- Add to source definition YAML/JSON

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
*closes #3940.*

## How
*Describe the solution*

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in output spec
- [ ] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [ ] Documentation updated 
    - [ ] README
    - [ ] CHANGELOG.md
    - [ ] Reference docs in the `docs/integrations/` directory.
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Review by Zirochkaa — APPROVED** [review]

**Review by keu — APPROVED** [review]

**Review by davinchia — APPROVED** [review]
Nice!

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/utils.py [hunk 1]
```diff
@@ -25,12 +25,12 @@
 from typing import Any, Mapping
 
 import pendulum
-from pendulum import DateTime
+from pendulum import datetime
 
 
 class Utils:
```
Comment (keu) [inline airbyte-integrations/connectors/source-google-ads/source_google_ads/utils.py:31]: it is better to move this inside stream, it is not an util in common sense

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/utils.py [hunk 2]
```diff
@@ -25,12 +25,12 @@
...
     @staticmethod
-    def get_date_params(stream_slice: Mapping[str, Any], cursor_field: str, end_date: DateTime = pendulum.yesterday()):
+    def get_date_params(stream_slice: Mapping[str, Any], cursor_field: str, end_date: datetime = pendulum.yesterday()):
```
Comment (keu) [inline airbyte-integrations/connectors/source-google-ads/source_google_ads/utils.py:33]: dangerous default `yesterday`

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py [hunk 3]
```diff
@@ -120,13 +129,17 @@ class AdGroupAdReport(IncrementalGoogleAdsStream):
 
 # Source
 class SourceGoogleAds(AbstractSource):
-    def check_connection(self, logger, config) -> Tuple[bool, any]:
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, any]:
         try:
             logger.info("Checking the config")
-            GoogleAds(**config)
+            check_config = dict(credentials=config["credentials"], customer_id=config["customer_id"])
+            GoogleAds(**check_config)
```
Comment (keu) [inline airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py:136]: use explicit params

File: airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py [hunk 4]
```diff
@@ -120,13 +129,17 @@ class AdGroupAdReport(IncrementalGoogleAdsStream):
...
-            return False, e
+        except Exception as error:
+            return False, f"Unable to connect to Google Ads API with the provided credentials - {repr(error)}"
 
     def streams(self, config: Mapping[str, Any]) -> List[Stream]:
-        return [AdGroupAdReport(config)]
+        stream_config = dict(
+            credentials=config["credentials"], customer_id=config["customer_id"], conversion_window_days=config["conversion_window_days"]
+        )
+        return [AdGroupAdReport(start_date=config["start_date"], **stream_config)]
```
Comment (keu) [inline airbyte-integrations/connectors/source-google-ads/source_google_ads/source.py:145]: lets create `GoogleAds` here and pass it as `api` parameters

File: airbyte-integrations/connectors/source-google-ads/unit_tests/test_google_ads.py [hunk 5]
```diff
@@ -75,10 +75,12 @@ def load_from_dict(config):
...
-    "client_id": "client_id",
-    "client_secret": "client_secret",
-    "refresh_token": "refresh_token",
+    "credentials": {
+        "developer_token": "developer_token",
+        "client_id": "client_id",
+        "client_secret": "client_secret",
+        "refresh_token": "refresh_token",
+    },
     "customer_id": "customer_id",
```
Comment (keu) [inline airbyte-integrations/connectors/source-google-ads/unit_tests/test_google_ads.py:84]: I guess start_date is required

**Review by keu — APPROVED** [review]
small comments, lgtm

File: airbyte-integrations/connectors/source-google-ads/unit_tests/test_google_ads.py (hunk 5 above)
Comment (yevhenii-ldv) [reply airbyte-integrations/connectors/source-google-ads/unit_tests/test_google_ads.py:84]: yes, but this test tests only connect for `GoogleAds` class,  and uses `SAMPLE_CONFIG` without `start_date` and `conversion_window_days`

**Pull request merged**